src/lib.rs

```diff
@@ -95,3 +97,9 @@ pub trait GenFn4<T>: Fn(&Seed, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&Seed, &Point2<T>) -> T {}
 impl<T, F> GenFn3<T> for F where F: Fn(&Seed, &Point3<T>) -> T {}
 impl<T, F> GenFn4<T> for F where F: Fn(&Seed, &Point4<T>) -> T {}
+
+pub trait Module<T> {
```

Comment:
Hmm... I'm actually thinking that these are really just functions. Perhaps we could use my `https://crates.io/crates/fn_ops` crate? We would then be implementing [`Fn<(T,)>`](http://bjz.github.io/fn_ops/fn_ops/trait.Fn.html) (see the [example](http://bjz.github.io/fn_ops/fn_ops/index.html#example)). The cool thing would be that you could then pass in functions or closures to the generators as well!